conda_lock/lockfile/v2prelim/models.py

```diff
@@ -17,20 +17,25 @@
 
 
 class LockedDependency(BaseLockedDependency):
-    def to_v1(self) -> LockedDependencyV1:
-        return LockedDependencyV1(
-            name=self.name,
-            version=self.version,
-            manager=self.manager,
-            platform=self.platform,
-            dependencies=self.dependencies,
-            url=self.url,
-            hash=self.hash,
-            category=self.category,
-            source=self.source,
-            build=self.build,
-            optional=self.category != "main",
-        )
+    categories: Set[str] = set()
```

Comment:
Shouldn't this be `set("main")`? The previous default value for `category` was `"main"`.

Reply:
I think you want `{"main"}`, as `set("main")` gives you `{"m", "a", ...`

Reply:
The problem is that in `apply_categories`, we assume that `categories` is initially empty and append to the set as needed. If we initially set it to `{'main'}`, then every dependency will be in the main category and will always be installed.